src/main/java/net/minecraftforge/client/event/InputEvent.java

```diff
@@ -199,4 +201,72 @@ public int getModifiers()
             return this.modifiers;
         }
     }
+
+    /**
+     * This event fires when one of the keybindings that by default involves clicking the mouse buttons
+     * is triggered.
+     *
+     * These key bindings are use item, pick block and attack keybindings. (right, middle and left mouse click)
+     * In the case that these key bindings are re-bound to a keyboard key the event will still be fired as normal.
+     */
+    @Cancelable
+    public static class ClickInputEvent extends InputEvent
+    {
+        private final int button;
+        private final KeyBinding keyBinding;
+        private final Hand hand;
+        private boolean handSwing = true;
+        public ClickInputEvent(int button, KeyBinding keyBinding, Hand hand)
+        {
+            this.button = button;
+            this.keyBinding = keyBinding;
+            this.hand = hand;
+        }
+
+        /**
+         * Set to false to disable the hand swing animation.
+         * @throws UnsupportedOperationException If this is not an attack input.
+         */
+        public void setSwingHand(boolean value)
+        {
+            if(!isAttack()) throw new UnsupportedOperationException("The hand swing can't be disabled if this is not an attack input.");
+            handSwing = value;
+        }
+
+        public boolean shouldSwingHand()
+        {
+            return handSwing;
+        }
+
+        /**
+         * The hand which is causing the event to get triggered.
+         * Will be null when this event was triggered by pick block as it has
+         * no associated hand.
+         * The event will be called for both hands if this is a use item input.
+         */
+        public Hand getHand()
+        {
+            return hand;
+        }
+
+        public boolean isAttack()
```

Comment:
I would personally name this isLeftClick(), isMiddleClick(), etc... as this contextually feels more appropriate. However I wouldn't really care too much.

Reply:
They were originally called isLeftClick(), isMiddleClick(), etc. but because you can rebind the keys that could cause confusion. (you could swap attack/use block which would swap isLeftClick() and isRightClick()) Calling after the keybindings probably minimises confusion.